### ISSUE TITLE
Unknown variable Accessed

### ISSUE THREAD

**Issue by laura-herrera** [wata727/tflint#168]
Variables are set in var files and those files are specified in tflint.config.  Yet, when running the lint, I get:
```
Evaluation error: 1:3: unknown variable accessed: var.environment 
```
Every time I run the lint, a different variable appears, all of which have been defined inside the vars file. All files are on the same directory.

### Configuration file
config {
  terraform_version = "0.11.2"

  aws_credentials = {
    access_key = "key"
    secret_key = "secret"
    region     = "us-east-1"
  }

  varfile = ["global.tfvars", "external.tfvars"]
}

### Globals - global.tfvars
variable "environment" {
  description = "Environment"
  default = "stg"
}

### Command
$ tflint --config tflint.config --deep *.tf -d

Output:
DEBUG:/loader.go:191 [INFO] Load HCL file: `tflint.config`
DEBUG:/loader.go:53 [INFO] Load HCL file: `tflint.config`
DEBUG:/loader.go:120 [INFO] Load environment...
DEBUG:/loader.go:128 [ERROR] open .terraform/environment: no such file or directory
DEBUG:/loader.go:131 [INFO] Load tfstate...
DEBUG:/loader.go:135 [ERROR] stat terraform.tfstate: no such file or directory
DEBUG:/loader.go:141 [INFO] Remote state detected
DEBUG:/loader.go:161 [INFO] Load tfvars...
DEBUG:/loader.go:164 [INFO] Load `terraform.tfvars`
DEBUG:/loader.go:166 [ERROR] stat terraform.tfvars: no such file or directory
DEBUG:/loader.go:164 [INFO] Load `global.tfvars`
DEBUG:/loader.go:191 [INFO] Load HCL file: `global.tfvars`
DEBUG:/loader.go:164 [INFO] Load `external.tfvars`
DEBUG:/loader.go:191 [INFO] Load HCL file: `external.tfvars`
DEBUG:/loader.go:191 [INFO] Load HCL file: `cluster.tf`
DEBUG:/loader.go:53 [INFO] Load HCL file: `cluster.tf`
Evaluation error: 1:3: unknown variable accessed: var.environment in cluster.tf:41

Terraform plan works perfectly fine, and apply also, the cluster is already built.
It's all the same if I use --deep, or if I don't.

Also, the same problem presents when giving the var files on the command line:
$ tflint --config tflint.config  *.tf --var-file=vaglobal.tfvars,external.tfvars -d

### Versions
$ tflint --version
TFLint version 0.5.4

$ terraform version
Terraform v0.11.2

**Comment by netflash** [comment]
same story here, but terraform version 0.11.3

**Comment by netflash** [comment]
debug run log

```
DEBUG:/loader.go:120 [INFO] Load environment...
DEBUG:/loader.go:128 [ERROR] open .terraform/environment: no such file or directory
DEBUG:/loader.go:131 [INFO] Load tfstate...
DEBUG:/loader.go:135 [ERROR] stat terraform.tfstate: no such file or directory
DEBUG:/loader.go:141 [INFO] Remote state detected
DEBUG:/loader.go:161 [INFO] Load tfvars...
DEBUG:/loader.go:164 [INFO] Load `terraform.tfvars`
DEBUG:/loader.go:191 [INFO] Load HCL file: `terraform.tfvars`
DEBUG:/loader.go:164 [INFO] Load `terraform.tfvars`
DEBUG:/loader.go:191 [INFO] Load HCL file: `terraform.tfvars`
DEBUG:/loader.go:191 [INFO] Load HCL file: `cloudtrail.tf`
DEBUG:/loader.go:53 [INFO] Load HCL file: `cloudtrail.tf`
Evaluation error: 1:3: unknown variable accessed: var.env in cloudtrail.tf:6
```

**Comment by tomqwu** [comment]
same error

**Comment by Stretch96** [comment]
I ran into this issue when there are unquoted values in the tfvars file.
Ensure all strings, bools and integers are quoted.

I believe this is still a bug though, as things should be allowed to be unquoted in tfvar files

**Comment by erikpaasonen** [comment]
This issue can be reproduced by passing an `aws_s3_bucket` attribute to a module as an input, e.g.
``` hcl
provider "aws" {
  region = "us-west-1"
}

data "aws_caller_identity" "current" {}

resource "aws_s3_bucket" "sample_bucket" {
  bucket = "my-${data.aws_caller_identity.current.account_id}-bucket"
  acl    = "log-delivery-write"
}

module "s3_bucket_ref" {
  source = "./module"
  s3_bucket = "${aws_s3_bucket.sample_bucket.bucket}"
}
```
and then inside `./module` place a file `vars.tf` with the following:
``` hcl
variable "s3_bucket" {}
```
In our case we have a hard requirement for the functionality our module provides to use the bucket, so it absolutely must exist already before the module runs. Since modules cannot use `depends_on`, we are simulating the dependency relationship by passing in an attribute of the bucket in to the module.

**Comment by yardensachs** [comment]
Any news on this issue?

**Comment by stefanverhoeff** [comment]
Experienced the same issue @Stretch96 mentioned: error shows up for tfvar that is a number. Quoting it to a string fixed the error. The terraform apply worked fine with either version.

TFLint version 0.7.2

**Comment by neechbear** [comment]
Also experiencing this with TFLint version 0.7.2.

*looks hopefully for the fix soon* :-) :+1:

**Comment by Stretch96** [comment]
@smiller171 Have you ensured you have `vpc_id` defined in a variable block too?

eg.

```
variable "vpc_id" {
  type = "map"
}
```

**Comment by smiller171** [comment]
@Stretch96 Yes. Terraform understands what's going on and I can deploy, it's only tflint I'm struggling with

**Comment by erikpaasonen** [comment]
@smiller171 how about refactoring to use `lookup()`? 
``` hcl
  vpc_id = "${lookup(var.vpc_id, terraform.workspace)}"
```
I haven't had much luck indexing map variables directly like that. Maybe tflint is giving you a legitimate smell.

**Comment by smiller171** [comment]
In my case it's definitely complaining about the variable in parent, not the module, based on the file name and line number it's giving me.

I would rather just not use tflint than refactor my code in that way since I find it much less readable, but I'll change it long enough to test

**Comment by smiller171** [comment]
Using a lookup does indeed solve the issue with tflint

**Comment by wata727** [comment]
@smiller171 Sorry for the inconvenience. This strange behavior will be fixed in 0.8.0. Until then, I encourage you to use a workaround or stop to use TFLint temporary...

**Comment by smiller171** [comment]
ETA on 0.8.0?

**Comment by wata727** [comment]
It depends on Terraform v0.12 release schedule. All tasks have been completed, but for that reason, I can't merge #209.

**Comment by cultavix** [comment]
We are getting a similar issue, except with the output from a module. 

`TFLINT_VERSION: "0.7.5"`

```
DEBUG:/loader.go:60 [INFO] Load HCL file: `vars_project.tf`
Evaluation error: 2:26: unknown variable accessed: module.sg_lifecycle.security_group_ids in main.tf:128
```

I'm going to have to disable.

**Comment by wata727** [comment]
Hi there, I just released TFLint v0.8.0 and fixed the weird behaviors. Please try the latest version again. If you still have problems, please open a new issue instead of reopening.
